docs/src/pages/guides/FHIRServerUsersGuide.md

```diff
@@ -1505,23 +1508,230 @@ To properly configure the FHIR server's keystore and truststore files, perform t
 
-### 5.2.5 OAuth 2.0
-In the default configuration, the FHIR server acts as an authorization server as well as a resource server. The FHIR server's conformance statement includes the OAuth-related URLs you will need to get started. The following excerpt from a conformance statement shows sample OAuth-related URLs (token, authorize, and register) as values of the `valueUri` elements.
+## 5.3 OpenID Connect and OAuth 2.0
+The FHIR specification recommends the use of OpenID Connect and OAuth 2.0.
+The IBM FHIR Server supports these via Liberty's OpenID Connect support.
+The following sections are adapted from https://www.ibm.com/support/knowledgecenter/SSD28V_liberty/com.ibm.websphere.wlp.core.doc/ae/twlp_config_oidc_pc_examp_beginner.html and related pages in the WebSphere Liberty knowledge center, but they apply to OpenLiberty
+as well.
+
+### 5.3.1 Configure Liberty as the OpenID Connect Provider
+Liberty can be configured to act as an OpenID Connect Provider via the [openidConnectServer-1.0 feature](https://openliberty.io/docs/ref/feature/#openidConnectServer-1.0.html).
+
+### 5.3.1.1 Configure Liberty as an OAuth 2.0 Provider
+OpenID Connect is built on OAuth 2.0 and so you must first configure Liberty as an OAuth provider as described at https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/twlp_oauth_defining.html.
+
+Liberty supports the registration of clients through either a localStore in the server.xml or a databaseStore in a configured dataSource.
+To support "dynamic client registration", use a databaseStore by:
 
+1. Configuring a clientManager oauth-role and specifying a databaseStore in the oauthProvider element of `server.xml`.
+    For example, to use Apache Derby for the store, include something like the following:
+    ```
+    <oauth-roles>
+        <authenticated>
+            <special-subject type="ALL_AUTHENTICATED_USERS"/>
+        </authenticated>
+        <clientManager>
+            <group name="clientAdministrator" />
+        </clientManager>
+    </oauth-roles>
+
+    <oauthProvider id="oauth2-provider" oauthOnly="false" allowPublicClients="true" jwtAccessToken="true">
+        <grantType>authorization_code</grantType>
+        <databaseStore dataSourceRef="OAuthDataSource" />
+    </oauthProvider>
+
+    <dataSource id="OAuthDataSource" jndiName="jdbc/OAuth2DB">
+        <properties.derby.embedded createDatabase="create" databaseName="derby/oauth2db" />
+        <jdbcDriver libraryRef="derbyLib"/>
+    </dataSource>
+
+    <library id="derbyLib">
+        <fileset dir="${shared.resource.dir}/lib/derby" includes="*.jar"/>
+    </library>
+    ```
+
+2. Manually creating the necessary tables in your database as described at https://www.ibm.com/support/knowledgecenter/SSD28V_liberty/com.ibm.websphere.wlp.core.doc/ae/twlp_oauth_dbs.html:
+
+```
```

Comment:
put this in a file in the `fhir-server`?

Reply:
yeah.  i still think its dumb that we need to create these tables at all...why doesn't it just auto-create the tables like the batch feature?
anyway, I need to decide how to package all this info to make it easier to configure.  I was thinking it would just be an optional configDropin, but then we have this stupid DDL that we need to run.  Maybe we just invoke it as part of fhir-persistence-schema even though its only used if you are using OAuth with a databaseStore...

Reply:
I've now modified `fhir-persistence-schema` to deploy these tables by default.
Additionally, I updated the derby bootstrapping to create them when the `jdbc/OAuth2DB` datasource is available.